File: Modules/SIE.ZYF/SIE.Web.ZYF/Materials/Commands/AddMaterialCommand.js
SIE.defineCommand("SIE.Web.ZYF.Materials.Commands.AddMaterialCommand", {
    extend: "SIE.cmd.Add",
    meta: { text: "添加", group: "edit" },
    onItemCreated: function (entity) {
        var model = entity.data;
        var me = this;
        me.view.execute({
            data: model,
            success: function (res) {
                entity.setCode(res.Result);
            }
        }, me.view);
    }
});

File: Modules/SIE.ZYF/SIE.Web.ZYF/Materials/Commands/AddMaterialCommand.cs
using SIE.Web.Command;
using SIE.ZYF.Materials;

namespace SIE.Web.ZYF.Materials.Commands
{
    public class AddMaterialCommand : ViewCommand
    {
        protected override object Excute(ViewArgs args, string scope)
        {
            var material = args.Data.ToJsonObject<Material>();
            material.Code = RT.Service.Resolve<MaterialsController>().GetCode();

            return material.Code;
        }
    }
}
